src/Core/Content/Media/Core/Application/RemoteThumbnailLoader.php

```diff
@@ -168,19 +173,37 @@ private function getBaseUrl(): string
         return \rtrim($this->filesystem->publicUrl(''), '/');
     }
 
-    private function getUrl(string $mediaUrl, string $mediaPath, string $width, string $height, ?\DateTimeInterface $mediaUpdatedAt): string
+    private function getUrl(MediaEntity|PartialEntity $mediaEntity, string $mediaUrl, string $width, string $height): ?string
     {
         return $this->extensions->publish(
             name: ResolveRemoteThumbnailUrlExtension::NAME,
             extension: new ResolveRemoteThumbnailUrlExtension(
                 $mediaUrl,
-                $mediaPath,
+                $mediaEntity->get('path'),
                 $width,
                 $height,
                 $this->pattern,
-                $mediaUpdatedAt
+                $mediaEntity->get('updatedAt') ?? $mediaEntity->get('createdAt'),
+                $mediaEntity,
             ),
-            function: function (string $mediaUrl, string $mediaPath, string $width, string $height, string $pattern, ?\DateTimeInterface $mediaUpdatedAt) {
+            function: function (
+                string $mediaUrl,
+                string $mediaPath,
+                string $width,
+                string $height,
+                string $pattern,
+                ?\DateTimeInterface $mediaUpdatedAt,
+                Entity $mediaEntity,
+            ): string {
+                /**
+                 * @deprecated tag:v6.8.0 - remove the nullish coalescing assignment when the property $mediaPath is removed from the extension
+                 */
+                $mediaPath ??= $mediaEntity->get('path');
```

Comment:
that is tricky now :grimacing: of course it cannot be `null` as you do not allow the type and it is previously set with `$mediaEntity->get('path')` before. 

You cannot simply use `$mediaEntity->get('path')` here, as this would break the extensions which set the `mediaPath` to something different. 

On the other hand if someone wants to be forward compatible, the path should be set via the MediaEntity and taken from there. 

maybe we need a condition with feature flag here. if 6.8 is not active use `mediaPath` directly. else use the value from the mediaEntity. 
to be forward compatible, plugins should not set both in their extension. 

What do you think?

Reply:
In my opinion, this is not a problem:
$mediaPath is not nullable, so the condition will never apply until the property is removed from the extension.
With $mediaUpdatedAt, it could happen if someone uses the extension to set $mediaUpdatedAt to null, which I don't think will ever happen.

Furthermore, the extension is not yet in SW 6.7, but only in the trunk.